x-pack/filebeat/input/etw/input.go

```diff
@@ -0,0 +1,296 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package etw
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	stateless "github.com/elastic/beats/v7/filebeat/input/v2/input-stateless"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/x-pack/libbeat/reader/etw"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"golang.org/x/sys/windows"
+)
+
+const (
+	inputName = "etw"
+)
+
+// It abstracts the underlying operations needed to work with ETW, allowing for easier
+// testing and decoupling from the Windows-specific ETW API.
+type ETWSessionOperator interface {
+	NewSession(config config) (*etw.Session, error)
+	AttachToExistingSession(session *etw.Session) error
+	CreateRealtimeSession(session *etw.Session) error
+	StartConsumer(session *etw.Session) error
+	StopSession(session *etw.Session) error
+}
+
+type RealETWSessionOperator struct{}
```

Comment:
Does this need to be exported? It looks like an internal-only thing for the input.